atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/Ec2HostLocationSupplier.java

```diff
@@ -46,13 +47,32 @@ public HostLocation get() {
             Preconditions.checkState(response.isSuccessful(),
                     "Getting AWS host metadata was not successful");
 
-            String responseString = response.body().string();
-            String datacenter = responseString.substring(0, responseString.length() - 2);
-            String rack = responseString.substring(responseString.length() - 1);
-
-            return HostLocation.of(datacenter, rack);
+            return parseHostLocation(response.body().string());
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
+
+    @VisibleForTesting
+    static HostLocation parseHostLocation(String az) {
+        // Code is copied from Cassandra's Ec2Snitch. The result of this parsing must match Cassandra's as closely as
```

Comment:
We should reimplement this method, as otherwise we'll need to go through the documentation for licensing (which is really too much work for this).